langtest/augmentation/__init__.py

```diff
@@ -241,3 +248,178 @@ def _parameters_overrides(self, config: dict, data_handler: List[Sample]) -> dic
                 .get("labels")
             )
         return config
+
+
+class TemplaticAugment(BaseAugmentaion):
+    """
+    This class is used for templatic augmentation. It is a subclass of the BaseAugmentation class.
+
+    Attributes:
+    __templates: A string or a list of strings or samples that represents the templates for the augmentation.
+    __task: The task for which the augmentation is being performed.
+
+    Methods:
+    __init__(self, templates: Union[str, List[str]], task: str): Initializes the TemplaticAugment class.
+    fix(self, input_path: str, output_path: str, *args, **kwargs): Performs the templatic augmentation and exports the results to a specified path.
+    """
+
+    def __init__(self, templates: Union[str, List[str]], task: str) -> None:
+        """
+        This constructor for the TemplaticAugment class.
+
+        Parameters:
+        templates (Union[str, List[str]]): The templates to be used for the augmentation.
+        task (str): The task for which the augmentation is being performed.
+        """
+        self.__templates: Union[str, List[str], List[Sample]] = templates
+        self.__task = task
+
+        if isinstance(self.__templates, str) and os.path.exists(self.__templates):
+            self.__templates = DataFactory(self.__templates, self.__task).load()
+        elif isinstance(self.__templates, str):
+            self.__templates = [self.str_to_sample(self.__templates)]
+        elif isinstance(self.__templates, list):
+            self.__templates = [self.str_to_sample(i) for i in self.__templates]
```

Comment:
the last condition should be `isinstance(self.__templates, list) and isinstance(self.__templates[0], str)` because it could be already be a `Sample`

Reply:
Can you fix the discrepancies between the arguments types and the types specified in the docs please